src/Data/Text/IO/Utf8.hs

```diff
@@ -0,0 +1,44 @@
+-- |
+-- Module      : Data.Text.IO.Utf8
+-- Copyright   : (c) 2009, 2010 Bryan O'Sullivan,
+--               (c) 2009 Simon Marlow
+-- License     : BSD-style
+-- Maintainer  : bos@serpentine.com
+-- Portability : GHC
+--
+-- Efficient UTF-8 support for text I\/O.
+module Data.Text.IO.Utf8
+    (
+      readFile
+    , writeFile
+    , appendFile
+    ) where
+
+import Prelude hiding (readFile, writeFile, appendFile)
+import Control.Exception (evaluate)
+import Control.Monad ((<=<))
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import Data.Text (Text)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+
+decodeUtf8IO :: ByteString -> IO Text
+decodeUtf8IO = evaluate . decodeUtf8
+
+encodeUtf8IO :: Text -> IO ByteString
+encodeUtf8IO = evaluate . encodeUtf8
```

Comment:
I'm curious if this is necessary. Is it possible for `encodeUtf8` to throw an exception? If so, is it possible for throwing at `evaluate` to differ from throwing at `B.writeFile` or `B.appendFile`?

Reply:
Yeah I don't think this is necessary